File: Núcleo/src/LogicaNegocio/Personal.java
package LogicaNegocio;

import Recursos.DatosPersonal;
import java.util.Date;

/**
 * @author Victor Javier
 * @version 1.0
 * @created 18-may.-2019 06:27:02 p. m.
 */
public abstract class Personal {

    protected String rfc;
    protected String numeroTelefono;
    protected String numeroPersonal;
    protected String nombre;
    protected String apellido;
    protected Date fechaNacimiento;
    protected char sexo;
    protected String turno;

    public Personal() {

    }

    public String getRfc() {
        return rfc;
    }

    public void setRfc(String rfc) {
        this.rfc = rfc;
    }

    public String getNumeroTelefono() {
        return numeroTelefono;
    }

    public void setNumeroTelefono(String numeroTelefono) {
        this.numeroTelefono = numeroTelefono;
    }

    public String getNumeroPersonal() {
        return numeroPersonal;
    }

    public void setNumeroPersonal(String numeroPersonal) {
        this.numeroPersonal = numeroPersonal;
    }

    public String getNombre() {
        return nombre;
    }

    public void setNombre(String nombre) {
        this.nombre = nombre;
    }

    public String getApellido() {
        return apellido;
    }

    public void setApellido(String apellido) {
        this.apellido = apellido;
    }

    public Date getFechaNacimiento() {
        return fechaNacimiento;
    }

    public void setFechaNacimiento(Date fechaNacimiento) {
        this.fechaNacimiento = fechaNacimiento;
    }

    public char getSexo() {
        return sexo;
    }

    public void setSexo(char sexo) {
        this.sexo = sexo;
    }

    public String getTurno() {
        return turno;
    }

    public void setTurno(String turno) {
        this.turno = turno;
    }

   public abstract DatosPersonal registrar();

    public abstract DatosPersonal modificar();

    /**
     *
     * @param hora
     * @param numeroConsultorio
     */
    public abstract boolean registrarEntrada(String numeroConsultorio);

    /**
     *
     * @param hora
     */
    public abstract boolean registrarSalida();

    /**
     *
     * @param estado
     */
    public abstract boolean eliminar();

    /**
     *
     * @param idUsuario
     */
    public abstract Personal obtenerPersonal(String numeroPersonal);
}//end Personal

File: Nucleo servidor web/application/libraries/Personal.php
<?php
abstract class Personal{
    private $rfc;
    private $numeroTelefono;
    private $numeroPersonal;
    private $nombre;
    private $apellido;
    private $fechaNacimiento;
    private $sexo;
    private $turno;
    private $estado;

    public function __get($attr) {
		return CI_Controller::get_instance()->$attr;
    }
    
    public function __construct(){
        
    }
    
    public function getRfc() {
        return $this->rfc;
    }

    public function setRfc($rfc) {
        $this->rfc = $rfc;
    }

    public function getNumeroTelefono() {
        return $this->numeroTelefono;
    }

    public function setNumeroTelefono($numeroTelefono) {
        $this->numeroTelefono = $numeroTelefono;
    }

    public function getNumeroPersonal() {
        return $this->numeroPersonal;
    }

    public function setNumeroPersonal($numeroPersonal) {
        $this->numeroPersonal = $numeroPersonal;
    }

    public function getNombre() {
        return $this->nombre;
    }

    public function setNombre($nombre) {
        $this->nombre = $nombre;
    }

    public function getApellido() {
        return $this->apellido;
    }

    public function setApellido($apellido) {
        $this->apellido = $apellido;
    }

    public function getFechaNacimiento() {
        return $this->fechaNacimiento;
    }

    public function setFechaNacimiento($fechaNacimiento) {
        $this->fechaNacimiento = $fechaNacimiento;
    }

    public function getSexo() {
        return $this->sexo;
    }

    public function setSexo($sexo) {
        $this->sexo = $sexo;
    }

    public function getTurno() {
        return $this->turno;
    }

    public function setTurno($turno) {
        $this->turno = $turno;
    }
    public function getEstado(){
        return $this->estado;
    }
    public function setEstado($estado){
        $this->estado = $estado;
    }

    public abstract function registrar($idUsuario);
    public abstract function modificar();
    public abstract function registrarEntrada($numeroConsultorio);
    public abstract function registrarSalida();
    public abstract function eliminar();
    public abstract function obtenerPersonal($rfc);
    public abstract function obtenerPersonalId($idUsuario);
    public abstract function obtenerPersonalRol($rol);
}